packages/integration-tests/projects/connect-popup/index.js

```diff
@@ -0,0 +1,370 @@
+            });
+            await wait(500)
+            await controller.send({
+                type: 'emulator-start',
+                wipe: true,
+            });
+            await wait(500)
+            await controller.send({
+                type: 'emulator-setup',
+                ...setup.device,
+            });
+            await wait(500)
+            await controller.send({ type: 'emulator-allow-unsafe-paths' });
+            await wait(500)
+
+            for (f of setup.fixtures) {
+
+                await controller.send({ type: 'bridge-stop' });
+                await wait(1000)
+                await controller.send({ type: 'bridge-start', version: '2.0.27' });
+                await wait(1000)
+
+
+                log(f.method, "start");
+
+                const screenshotsPath = `${SCREENSHOTS_DIR}/${f.method}`;
+
+                if (!fs.existsSync(screenshotsPath)) {
+                    fs.mkdirSync(screenshotsPath);
+                }
+
+                const explorer = await browser.newPage();
+                const popupPromise = new Promise(x => explorer.once('popup', x));
+
+                await explorer.goto(`${url}#/method/${f.method}`);
+
+                // screenshot request
+                log(f.method, "screenshot trezor-connect call params");
+                await explorer.waitForTimeout(2000);
+                await explorer.screenshot({ path: `${screenshotsPath}/1-request.png`, fullPage: true });
+
+                log(f.method, "submitting in connect explorer");
+                await explorer.waitForSelector("button[data-test='@submit-button']", { visible: true });
+                await explorer.click("button[data-test='@submit-button']");
+
+                log(f.method, "waiting for popup promise");
+                const popup = await popupPromise;
+                log(f.method, "popup promise resolved");
+
+                // can't really connect device here, experiencing unexpected popup closing
+                // await popup.screenshot({ path: `${screenshotsPath}/2-connect-device.png`, fullPage: true });
+                // await popup.waitForSelector(".connect", { visible: true, timeout: 30000 });
+
+                try {
+                    log(f.method, "waiting for confirm permissions button");
+                    await popup.waitForTimeout(1000);
+                    await popup.waitForSelector("button.confirm", { visible: true, timeout: 30000 });
+                    await popup.screenshot({ path: `${screenshotsPath}/2-permissions.png`, fullPage: true });
+                    await popup.waitForTimeout(1000);
+
+                    await popup.click("button.confirm");
+                } catch (err) {
+                    log(f.method, "permissions button not found");
+                    await popup.screenshot({ path: `${screenshotsPath}/2-err-permissions-not-found.png`, fullPage: true });
+                }
+
+                let viewIndex = 0;
+                for (v of f.views) {
```

Comment:
In my opinion naming `v` is unclear and does not help readability, what about `view`?